### ISSUE TITLE
Adhere to the user's desktop environment's `QStyle`.

### ISSUE THREAD

**Issue by RokeJulianLockhart** [rmcrackan/Libation#765]
#### **Is your feature request related to a problem? Please describe.**

I use the Breeze `QStyle`:

1. `breeze-icon-theme-6.11.0-1.fc41.noarch`
1. `kf6-breeze-icons-6.11.0-1.fc41.x86_64`
1. `breeze-gtk-common-6.3.1-1.fc41.noarch`
1. `plasma-breeze-common-6.3.1-1.fc41.noarch`
1. `plasma-breeze-qt6-6.3.1-1.fc41.x86_64`
1. `plasma-breeze-qt5-6.3.1-1.fc41.x86_64`
1. `plasma-breeze-6.3.1-1.fc41.x86_64`
1. `qqc2-breeze-style-6.3.1-1.fc41.x86_64`
1. `breeze-cursor-theme-6.3.1-1.fc41.noarch`
1. `breeze-gtk-gtk4-6.3.1-1.fc41.noarch`
1. `breeze-gtk-gtk3-6.3.1-1.fc41.noarch`
1. `breeze-gtk-gtk2-6.3.1-1.fc41.noarch`

...per `kcm_style`:

![Image](https://github.com/user-attachments/assets/fdbe5930-5f68-4815-8503-2e7a77e6e33b)

However, `libation-12.0.2-1.x86_64` clearly doesn't care:

![Image](https://github.com/user-attachments/assets/4839348b-61b2-4266-8114-d33a16e321a3)

I don't know what GUI toolkit Libation utilises, but even if it's GTK3/4, it'd work:

![Image](https://github.com/user-attachments/assets/3227156b-5064-40e5-91e7-7ff48e368e6c)

#### **Describe the solution you'd like**

> Adhere to the user's desktop environment's `QStyle`.
	
#### **Describe alternatives you've considered**

Rewrite the GUI.
	
#### **Additional context**

I understand that if you're not using Qt, this might not be achievable.

**Comment by Mbucari** [comment]
@RokeJulianLockhart The problem is that Libation is _not_ a native linux app, or even specifically targeted at Linux. Chardonnay uses a cross-platform GUI framework that inherits only the window border "decorations" (i.e. the title bar) from the system's GUI. The rest of its look must be defined and designed by the developer (in this case, me), so it will always look out of place on someone's system because it looks the same on every system.

So I'm sorry, but this is just not in the cards.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/rmcrackan/Libation/issues/765#issuecomment-2701236819

@Mbucari, according to [`reddit.com/r/audible/comments/w9xp8i`](https://www.reddit.com/r/audible/comments/w9xp8i/libation_chardonnay_v83_linux_beta/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button), am I correct that your GUI toolkit is custom? If so, that's more understandable. I'm glad that you at least keep the window server-side decorated.

**Comment by Mbucari** [comment]
> am I correct that your GUI toolkit is custom?

Yes. It's [Avalonia UI](https://avaloniaui.net/) which is basically [Windows Presentation Foundation](https://en.wikipedia.org/wiki/Windows_Presentation_Foundation) made cross-platform. It comes with two theme variants: "light" (aka "default") and "dark", and nearly all of the colors come from [the themes resource file](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Accents/BaseColorsPalette.xaml).

> note that the current colouration would be problematic for those with astigmatisms.
Is that because it's black, not just a dark gray? If so, _that_ is a much more tractable issue. _Someone_ may be able to add a "gray" theme.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/rmcrackan/Libation/issues/765#issuecomment-2701323300

I wouldn't call Avalonia "custom". It's a very widely-used toolkit, as you presumably well know.

Regardless, per https://github.com/AvaloniaUI/Avalonia/pull/8148#event-6622189668, it looks like there's fairly good support for native platform themes for Windows, but not elsewhere. I certainly shouldn't expect one toolkit to inherit from another (this isn't wxWidgets).

Thanks.

> *that* is a much more tractable issue. *Someone* may be able to add a "gray" theme.

...Does https://github.com/rmcrackan/Libation/issues/766#issue-1912356850 count, or should it also be closed? In retrospect, it seems like I'd already ascertained the toolkit when I filed that issue, since https://github.com/rmcrackan/Libation/issues/766#issuecomment-2409096908 mentions some potential ways to achieve that in Avalonia.

**Comment by Mbucari** [comment]
> it looks like there's fairly good support for native platform themes for Windows, but not elsewhere.

Not really. Since Avalonia's herritage is WPF, it may inherrit a bit of the "look", but the themes are stand-alone collections of colors, fonts, shapes, and layouts. That PR you linked to is just about moving the themes to external repositories which must be loaded by the app, ads Libation does here:
https://github.com/rmcrackan/Libation/blob/b051283fcaa3df8b158e3e5fc0ba8be9dbd404cb/Source/LibationAvalonia/App.axaml#L67-L70

The "Fluent" themes were created from scratch by the Avalonia team, and include only the "Light" and "Dark" variants. They do not inherit from the host system. As discussed [in this post](https://github.com/AvaloniaUI/Avalonia/discussions/11953#discussioncomment-6319425), they do not plan to add any other variants. The best we could do here is override some of the color pallet choices in the "Dark" variant.

> ...Does [#766 (comment)](https://github.com/rmcrackan/Libation/issues/766#issue-1912356850) count, or should it also be closed?

Needs to be closed. Any solution that could "inherit" would be highly system-specific. The very best we could do is make the theme user-customizable, and that would be a herculean effort.

> In retrospect, it seems like I'd already ascertained the toolkit when I filed that issue, since [#766 (comment)](https://github.com/rmcrackan/Libation/issues/766#issuecomment-2409096908) mentions some potential ways to achieve that in Avalonia.

You linked to two Avalonia issues. The first issue regards nothing more than adding the existing "Light" and "Dark" themes, and detection for whether the host is set to a "Light" or "Dark" theme to make the Avalonia app match. Libation already does this. If you set the theme to "Default", it chooses either "Light" or "Dark" depending on the host settings. But they're the same "Light" and "Dark" and look the same on every platform.

The second issue you linked is merely about getting the accent color _used by the current theme_ (which again, is either "Light" or "Dark"). Now, Accent Colors _may_ (and usually _do_) come from the system's settings via [PlatformSettings.GetColorValues()](https://docs.avaloniaui.net/docs/concepts/services/platform-settings#getcolorvalues), but that is **only** the accent color. All other colors are defined at the theme-level and are independent of platform.

**Comment by RokeJulianLockhart** [comment]
> The "Fluent" themes were created from scratch by the Avalonia team, and include only the "Light" and "Dark" variants. They do not inherit from the host system.

@Mbucari, that's what I meant. If ever you've tried to read the accent colour values from the Registry, you'll know that it's not really possible to inherit most theme values from the OS after XP (and even its theming engine required a tweak to enable).
